code/src/java/pcgen/output/publish/OutputDB.java

```diff
@@ -206,6 +215,68 @@ public static boolean isLegal(String interpolation)
 	public static void reset()
 	{
 		outModels.clear();
+		globalModels.clear();
+		triggerLoad();
+	}
+
+	/**
+	 * Returns a map of the global TemplateModel objects (those that do not
+	 * depend on a PC)
+	 * 
+	 * Note that ownership of the returned map is transferred to the calling
+	 * object, no changes to the returned map will impact OutputDB, nor will
+	 * changes to OutputDB impact the returned Map.
+	 * 
+	 * @return a Map of the global TemplateModel objects
+	 */
+	public static Map<String, TemplateModel> getGlobal()
+	{
+		Map<String, TemplateModel> hm = new HashMap<String, TemplateModel>();
+		for (NamedModel nm : globalModels.values())
+		{
+			hm.put(nm.getModelName(), nm);
+		}
+		return hm;
+	}
+
+	/**
+	 * Registers a new Boolean Preference for inclusion in the global Models.
+	 * 
+	 * @param pref
+	 *            The preference name, as identified in the preference file
+	 * @param defaultValue
+	 *            The default value for the preference if it is not defined
+	 */
+	public static void registerBooleanPreference(String pref,
+		boolean defaultValue)
+	{
+		if ((pref == null) || (pref.length() == 0))
+		{
+			throw new IllegalArgumentException(
+				"Preference Name may not be null or empty: " + pref);
+		}
+		BooleanOptionModel bom = new BooleanOptionModel(pref, defaultValue);
+		TemplateModel old = globalModels.put(pref, bom);
+		if (old != null)
+		{
+			throw new UnsupportedOperationException(
+				"Cannot have two Preference Output Models using the same name: "
+					+ pref);
+		}
+	}
+
+	/*
+	 * Initialize the preferences we want to export to FreeMarker
+	 */
+	static
+	{
+		triggerLoad();
+	}
+
+	private static void triggerLoad()
+	{
+		registerBooleanPreference(
```

Comment:
I'm worried about the order of operations as this is run from a static. Have the options been loaded by that stage of program startup? In general I like to avoid static initialisation, particularly if it has external dependencies - you lose a lot of control over the  order of things. I would generally setup this sort of object as a singleton instance (perhaps managed by spring, or some master object) to allow for easier ordering and testing.